lib/build.ts

```diff
@@ -104,6 +105,20 @@ export function runBuildTask(
 		{ registryconfig: registrySecrets },
 		task.dockerOpts,
 	);
+
+	// Add the target platform to the build options if
+	// we build against a docker daemon that supports it.
+	if (
+		task.dockerPlatform &&
+		task.builderVersion &&
+		semver.satisfies(
+			semver.coerce(task.builderVersion.ApiVersion) ?? '0.0.0',
+			'>=1.38.0',
+		)
+	) {
+		task.dockerOpts = { platform: task.dockerPlatform, ...task.dockerOpts };
```

Comment:
There is already a `task.dockerOpts = _.merge(...)` call above, which by the way unnecessarily uses `lodash`. Suggestion:
```
	const usePlatform: boolean =
		task.dockerPlatform &&
		task.builderVersion &&
		semver.satisfies(
			semver.coerce(task.builderVersion.ApiVersion) ?? '0.0.0',
			'>=1.38.0',
		);

	task.dockerOpts = {
		registryconfig: registrySecrets,
		...(usePlatform ? { platform: task.dockerPlatform } : {}),
		...task.dockerOpts,
	};
```